src/common/mem.c

```diff
@@ -109,25 +130,36 @@ umempobj_create(const char *path, const char *layout_name, int flags,
 			}
 		}
 
-		if (store != NULL)
-			umm_pool->up_store = *store;
 		umm_pool->up_priv = pop;
 		return umm_pool;
-	}
+	case DAOS_MD_BMEM:
+		dav_hdl = dav_obj_create(path, 0, poolsize, mode);
+		if (!dav_hdl) {
+			D_ERROR("Failed to create pool %s, size="DF_U64": errno = %d\n",
+				path, poolsize, errno);
+			D_FREE(umm_pool);
+			return NULL;
+		}
+		umm_pool->up_priv = dav_hdl;
 
-	dav_hdl = dav_obj_create(path, 0, poolsize, mode);
-	if (!dav_hdl) {
-		D_ERROR("Failed to create pool %s, size="DF_U64": errno = %d\n",
-			path, poolsize, errno);
-		D_FREE(umm_pool);
-		return NULL;
-	}
-	if (store != NULL)
-		umm_pool->up_store = *store;
-	umm_pool->up_priv = dav_hdl;
+		/* TODO: Do checkpoint here to write back allocator heap */
```

Comment:
I think this is for both BMEM and AD? Or ad_blob_create() will internally flush heap data to SSD (and doesn't need to rely on WAL)?

Reply:
I don't quite understand the original comment for BMEM here.
For AD, it internally call store->stor_ops->so_write() to write the ARENA HDR + BLOB HDR in ad_blob_create().